### PR TITLE
Give the Spaghetti syntax a facelift

### PR DESCRIPTION
**Pull request by knuton**
This brings the language syntax closer to our main target languages and improves the legibility of method definitions.

On the one hand, we want the syntax to follow the main conventions of Haxe and TypeScript in ordering names and type annotations. On the other hand, we don't align syntax too closely with either language in order to provide signifiers that help prevent confusion about which language is currently being edited.

### CODE REVIEW

File: spaghetti-haxe-support/src/test/groovy/com/prezi/spaghetti/haxe/impl/HaxeModuleProxyGeneratorVisitorTest.groovy [hunk 1]
```diff
@@ -5,28 +5,29 @@ import com.prezi.spaghetti.generator.ModuleGeneratorSpecification
...
+		/**
+		 * This should have nothing to do with the results.
+		 */
+		someDummyMethod(x: int): void;
+	}
 	/**
-	 * This should have nothing to do with the results.
+	 * Does something.
 	 */
-	void someDummyMethod(int x)
```
Comment (lptr) [inline spaghetti-haxe-support/src/test/groovy/com/prezi/spaghetti/haxe/impl/HaxeModuleProxyGeneratorVisitorTest.groovy]: Why remove this?

File: spaghetti/ffs.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env ruby
+foo = ARGF.read
+puts foo.gsub(/\/\*([^*])/, '/**~SING \1').gsub(/\/\/(.*)/, '/**~LINE \1*/')
```
Comment (lptr) [inline spaghetti/ffs.rb]: I'd put these into a `migration` directory or something, instead of littering the root directory.

**Comment by lptr** [comment]
Seems like a lot of work, but looks good. :)

File: spaghetti-haxe-support/src/test/groovy/com/prezi/spaghetti/haxe/impl/HaxeModuleProxyGeneratorVisitorTest.groovy (hunk 1 above)
Comment (knuton) [inline spaghetti-haxe-support/src/test/groovy/com/prezi/spaghetti/haxe/impl/HaxeModuleProxyGeneratorVisitorTest.groovy]: I think the diff is just confusing, this is the interface method, it's added in the new syntax a few lines above.

**Comment by knuton** [comment]
I wasn't that bad! Finding my way around now ;)

**Pull request merged**